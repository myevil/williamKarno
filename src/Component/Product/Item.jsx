import React, { Component } from 'react'

class Item extends Component {
  handleAddCart(dataProduct) {
    dataProduct.total = 0;
    dataProduct.quantity = 0;
    this.props.handleAddShoppingCartData(dataProduct);
  }

  render() {
    const {data} = this.props;
    const {image, name, price, currency} = data;
    return (
      <div className={"scss-list-item"}>
        <div className={"border"}>
          <img src={image} alt='list-product'/>
        </div>
        <div className={'display'}>
          <div className={'grow'}>
            <p className={'product-title'}>{name}</p>
            <span>{currency} {price}</span>
          </div>
          <button
            className="product-button"
            onClick={() => this.handleAddCart(data)}
          >
            Add to cart
          </button>
        </div>
      </div>
    )
  }
}

export default Item
