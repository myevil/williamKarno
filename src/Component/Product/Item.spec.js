import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Item from './Item';

configure({ adapter: new Adapter() });

describe('test hompage', () => {
  const wrapper = shallow(<Item
    data={
      {
        data: {
          image: '',
          name: 'clothes',
          price: '25',
          currency: 'SGD',
          quantity: 0,
          total: 0,
        }
      }
    }
  />)
  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  })
})