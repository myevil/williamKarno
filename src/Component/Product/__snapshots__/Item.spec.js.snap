// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test hompage should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Item
    data={
      Object {
        "data": Object {
          "currency": "SGD",
          "image": "",
          "name": "clothes",
          "price": "25",
          "quantity": 0,
          "total": 0,
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="border"
        >
          <img
            alt="list-product"
          />
        </div>,
        <div
          className="display"
        >
          <div
            className="grow"
          >
            <p
              className="product-title"
            />
            <span>
               
            </span>
          </div>
          <button
            className="product-button"
            onClick={[Function]}
          >
            Add to cart
          </button>
        </div>,
      ],
      "className": "scss-list-item",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <img
            alt="list-product"
          />,
          "className": "border",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "list-product",
            "src": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="grow"
            >
              <p
                className="product-title"
              />
              <span>
                 
              </span>
            </div>,
            <button
              className="product-button"
              onClick={[Function]}
            >
              Add to cart
            </button>,
          ],
          "className": "display",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <p
                  className="product-title"
                />,
                <span>
                   
                </span>,
              ],
              "className": "grow",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                  "className": "product-title",
                },
                "ref": null,
                "rendered": null,
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    undefined,
                    " ",
                    undefined,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  undefined,
                  " ",
                  undefined,
                ],
                "type": "span",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Add to cart",
              "className": "product-button",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Add to cart",
            "type": "button",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="border"
          >
            <img
              alt="list-product"
            />
          </div>,
          <div
            className="display"
          >
            <div
              className="grow"
            >
              <p
                className="product-title"
              />
              <span>
                 
              </span>
            </div>
            <button
              className="product-button"
              onClick={[Function]}
            >
              Add to cart
            </button>
          </div>,
        ],
        "className": "scss-list-item",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <img
              alt="list-product"
            />,
            "className": "border",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "list-product",
              "src": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="grow"
              >
                <p
                  className="product-title"
                />
                <span>
                   
                </span>
              </div>,
              <button
                className="product-button"
                onClick={[Function]}
              >
                Add to cart
              </button>,
            ],
            "className": "display",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <p
                    className="product-title"
                  />,
                  <span>
                     
                  </span>,
                ],
                "className": "grow",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": undefined,
                    "className": "product-title",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      undefined,
                      " ",
                      undefined,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    undefined,
                    " ",
                    undefined,
                  ],
                  "type": "span",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Add to cart",
                "className": "product-button",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Add to cart",
              "type": "button",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
